tensorflow/core/ops/string_ops.cc

```diff
@@ -280,4 +280,32 @@ input: Base64 strings to decode.
 output: Decoded strings.
 )doc");
 
+REGISTER_OP("Substr")
```

Comment:
I'm a little dubious about this op as written -- I can see the use case if all inputs were scalars, but applying pos/len scalars to all elements of an input seems potentially limiting.

I was maybe thinking that pos and len could be equal-shaped tensors, and have pos and len be broadcastable against 'input'.  For example, if they were scalars, then they would apply to every element.

If the shape of 'pos' and 'len' matched 'input', then the entries in pos and len would be elementwise applied to input.  Otherwise, they would follow our standard broadcasting rules.  (e.g., you could have a matrix [MxN] of strings, have pos and len be [1xN], and then pos and len would apply row-wise to every element of the input matrix).

From an API perspective, that seems more flexible, even if for now, you only implement the scalar or equal-shaped versions and threw errors for the rest.